### PR TITLE
Support WebSocket for Binance Spot trading

### PR DESCRIPTION
**Pull request by trexnix**
Support WebSocket for Binance's [WebSocket streams](https://github.com/binance-exchange/binance-official-api-docs/blob/master/web-socket-streams.md) and [User Data stream](https://github.com/binance-exchange/binance-official-api-docs/blob/master/user-data-stream.md)

Usage example:

```
defmodule A do
  use Binance.WebSocket.WSClient
end

# Public channels/streams
A.start_link(%{name: :"btcusdt-depth-stream", public_channels: ["btcusdt@depth"]})

# User Data stream
A.start_link(%{name: :"user-data-stream", require_auth: true})

### CODE REVIEW

File: lib/binance/rest/http_client.ex [hunk 1]
```diff
@@ -61,30 +63,37 @@ defmodule Binance.Rest.HTTPClient do
...
-      :crypto.hmac(
-        :sha256,
-        Application.get_env(:binance, :secret_key),
-        argument_string
-      )
-      |> Base.encode16()
+    case HTTPoison.post("#{@endpoint}#{url}", body, [
+           {"X-MBX-APIKEY", Application.get_env(:binance, :api_key)},
+           {"Content-Type", "application/x-www-form-urlencoded"}
+         ]) do
```
Comment (sirius248) [inline lib/binance/rest/http_client.ex]: This whole thing can be wrap inside a function also. Maybe under `Util`

```elixir
[
           {"X-MBX-APIKEY", Application.get_env(:binance, :api_key)},
           {"Content-Type", "application/x-www-form-urlencoded"}
]
```

File: lib/binance/util.ex [hunk 2]
```diff
@@ -0,0 +1,35 @@
+defmodule Binance.Util do
+  @moduledoc false
+
+  @doc """
+  Prepare request body, sign it if required
+  """
+  def prepare_request_body(params, sign?) do
```
Comment (sirius248) [inline lib/binance/util.ex]: I would suggest use `opts` for example: 

`prepare_request_body(params, opts \\ [])` => more descriptive. So when I use the function I have to use something like

`Util.prepare_request_body(params, [signed: true])`

File: lib/binance/util.ex [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+      false ->
+        argument_string
+    end
+  end
+
+  @doc """
+  Sign a value using Binance Secret key
+  """
+  def sign_content(content) do
```
Comment (sirius248) [inline lib/binance/util.ex]: only use here. So maybe `defp`

**Comment by trexnix** [comment]
I updated @kimquy

**Review by sirius248 — APPROVED** [review]
LGTM

**Pull request merged**